src/node/node_state.h

```diff
@@ -629,6 +629,10 @@ namespace ccf
               "Node has now joined the network as node {}: {}",
               self,
               (resp.network_info->public_only ? "public only" : "all domains"));
+
+            // TODO: Remove this before committing
+            LOG_FAIL_FMT("Sleeping for an extremely long time!");
+            std::this_thread::sleep_for(std::chrono::milliseconds(200));
```

Comment:
It's a timing issue, and only appears when the old node makes multiple initiation attempts immediately after we've joined the network (and are ready to process incoming channels). This is a temporary hack to try and get consistent repros - sleep at this point to encourage multiple attempts by the peer.